uniffi_bindgen/src/bindings/python/templates/RustBufferHelper.py

```diff
@@ -2,23 +2,47 @@
 class FfiConverterPrimitive:
     @classmethod
     def check(cls, value):
-        pass
+        return value
 
     @classmethod
     def lift(cls, value):
         return value
 
     @classmethod
     def lower(cls, value):
-        cls.check(value)
+        return cls.lowerUnchecked(cls.check(value))
+
+    @classmethod
+    def lowerUnchecked(cls, value):
         return value
 
+    @classmethod
+    def write(cls, value, buf):
+        cls.writeUnchecked(cls.check(value), buf)
+
 class FfiConverterPrimitiveInt(FfiConverterPrimitive):
     @classmethod
     def check(cls, value):
+        try:
+            value = value.__index__()
+        except Exception:
+            raise TypeError("'{}' object cannot be interpreted as an integer".format(type(value).__name__))
```

Comment:
As an aside, I'd probably be in favour of declaring we only work with 3.6 and up (even 3.7 TBH, seeing 3.6 is getting no security updates) which would mean f-strings could be used.

Reply:
side-note: Glean supports 3.6 and that's also the minimum for mozilla-central.